src/Domain/Tags.hpp

```diff
@@ -259,13 +256,23 @@ struct evaluate_compute_item<DirectionsTag, BaseComputeItem,
   }
 };
 
+template <typename DirectionsTag, typename Tag, typename = cpp17::void_t<>>
+struct InterfaceBaseTag {};
```

Comment:
Sorry, I'm obviously still confused on something. I'm going to say what I think is happening, and then please correct me where I'm wrong:

- The empty case, line 260, is for when there is no base class and so there is base tag mechanism active.
- The second case, line 263, is used if the tag as a `base` type alias, and is used to make `InterfaceImpl` inherit off of `Tag::base`. This means that `InterfaceBaseTag` is never used as a base tag, but is purely a method of getting the correct base class onto the interface tag.

If `InterfaceBaseTag` is only used for generating the inheritance maybe it should be named something like `GetBaseTagIfPresent`? I think with that sort of a name change this code would be easier to understand, since initially I was expecting the class itself to be used as a tag.

One small note:
The one thing I'm not sure on is if this will work with a `db::BaseTag`. Do you want it to? If not, it would be useful to state that, if so, it should be tested.

Reply:
Your understanding of what is going on is correct, and I like your renaming suggestion.

It does not currently work with `db::BaseTag` and I was going to say that there was a good reason for that, but now I think I know how to fix it.